okio-files/src/commonMain/kotlin/okio/Filesystem.kt

```diff
@@ -15,6 +15,74 @@
  */
 package okio
 
+/**
+ * Read and write access to a hierarchical collection of files, addressed by [paths][Path]. This
+ * is a natural interface to the [current computer's local filesystem][SYSTEM].
+ *
+ * Not Just the Local Filesystem
+ * -----------------------------
+ *
+ * Other implementations are possible:
+ *
+ *  * `FakeFilesystem` is an in-memory filesystem suitable for testing. Note that this class is
+ *    included in the `okio-testing` artifact.
+ *
+ *  * A ZIP filesystem could provide access to the contents of a `.zip` file.
+ *
+ *  * A remote filesystem could access files over the network.
+ *
+ *  * A decorating filesystem could apply monitoring, encryption, compression, or filtering to
+ *    another filesystem implementation.
+ *
+ * For improved capability and testability, consider structuring your classes to dependency inject
+ * a `Filesystem` rather than using [SYSTEM] directly.
+ *
+ * Limited API
+ * -----------
+ *
+ * This interface is limited in which filesystem features it supports. Applications that need rich
+ * filesystem features should use another API, possibly alongside this API.
```

Comment:
Is it worth having a clean example of this escape hatch? i.e. how would you temporarily access the JDK Path or File objects when you want to opportunistically call those methods?

Reply:
I don't really want to get into those examples because it gets ugly fast. The first thing you lose is testability!